nifi-extension-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/JWTBearerOAuth2AccessTokenProvider.java

```diff
@@ -0,0 +1,664 @@
+        tokenEndpoint = context.getProperty(TOKEN_ENDPOINT).getValue();
+        algorithmName = context.getProperty(SIGNING_ALGORITHM).getValue();
+        headerX5T = context.getProperty(HEADER_X5T).asBoolean();
+        grantType = context.getProperty(GRANT_TYPE).evaluateAttributeExpressions().getValue();
+        assertion = context.getProperty(ASSERTION).evaluateAttributeExpressions().getValue();
+
+        if (context.getProperty(ISSUER).isSet()) {
+            issuer = context.getProperty(ISSUER).evaluateAttributeExpressions().getValue();
+        } else {
+            issuer = null;
+        }
+
+        if (context.getProperty(SUBJECT).isSet()) {
+            subject = context.getProperty(SUBJECT).evaluateAttributeExpressions().getValue();
+        } else {
+            subject = null;
+        }
+
+        if (context.getProperty(AUDIENCE).isSet()) {
+            audience = context.getProperty(AUDIENCE).evaluateAttributeExpressions().getValue();
+        } else {
+            audience = null;
+        }
+
+        if (context.getProperty(SCOPE).isSet()) {
+            scope = context.getProperty(SCOPE).evaluateAttributeExpressions().getValue();
+        } else {
+            scope = null;
+        }
+
+        if (context.getProperty(JTI).isSet()) {
+            jti = context.getProperty(JTI).evaluateAttributeExpressions().getValue();
+        } else {
+            jti = null;
+        }
+
+        if (context.getProperty(KID).isSet()) {
+            kid = context.getProperty(KID).evaluateAttributeExpressions().getValue();
+        } else {
+            kid = null;
+        }
+
+        if (context.getProperty(SSL_CONTEXT_PROVIDER).isSet()) {
+            SSLContextProvider sslContextProvider = context.getProperty(SSL_CONTEXT_PROVIDER).asControllerService(SSLContextProvider.class);
+            keyManager = sslContextProvider.createKeyManager().orElseThrow(() -> new IllegalStateException("KeyManager not available"));
+        }
+
+        customClaims = new HashMap<>();
+        formParams = new HashMap<>();
+        for (PropertyDescriptor descriptor : context.getProperties().keySet()) {
+            if (descriptor.isDynamic()) {
+                if (descriptor.getName().startsWith(CLAIM_PREFIX)) {
+                    customClaims.put(StringUtils.substringAfter(descriptor.getName(), CLAIM_PREFIX),
+                            context.getProperty(descriptor).evaluateAttributeExpressions().getValue());
+                } else if (descriptor.getName().startsWith(FORM_PREFIX)) {
+                    formParams.put(StringUtils.substringAfter(descriptor.getName(), FORM_PREFIX),
+                            URLEncoder.encode(context.getProperty(descriptor).evaluateAttributeExpressions().getValue(), StandardCharsets.UTF_8));
+                }
+            }
+        }
+    }
+
+    private void initJWTSigner() {
+        JWSAlgorithm algorithm = JWSAlgorithm.parse(algorithmName);
+
+        if (privateKey instanceof RSAPrivateKey) {
```

Comment:
The cast variable name can be used, instead of casting below.
```suggestion
        if (privateKey instanceof RSAPrivateKey rsaPrivateKey) {
```